### PR TITLE
Service details not yet available in GovCloud.

### LINKED ISSUE #3506: v1.9.0: VPC Endpoint Data Resource no longer works in GovCloud
As of v1.9.0 of the terraform aws provider, I can no longer use the `aws_vpc_endpoint_service` data resource to get the `service_name` for the S3 VPC Endpoint in a GovCloud region. If I revert back to v1.8.0, it works fine.

### Terraform Version

```
> terraform -v
Terraform v0.11.3
```

### Affected Resource(s)
- `aws_vpc_endpoint_service`

### Terraform Configuration Files
```hcl
provider "aws" {
  version = "1.9.0"
}

data "aws_vpc_endpoint_service" "s3" {
  service = "s3"
}

resource "aws_vpc_endpoint" "s3" {
  vpc_id       = "vpc-foovpcid"
  service_name = "${data.aws_vpc_endpoint_service.s3.service_name}"
}
```

### Debug Output

[Debug output](https://gist.github.com/lorengordon/d5922f09165f8a0da0a33d7e4f6ef65b)

Gist contains both the non-working behavior in v1.9.0 and the working behavior in v1.8.0.

### Expected Behavior
Expected the data resource to resolve and provide the `service_name`, as it does in v1.8.0.

```
> terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.aws_vpc_endpoint_service.s3: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_vpc_endpoint.s3
      id:                <computed>
      cidr_blocks.#:     <computed>
      policy:            <computed>
      prefix_list_id:    <computed>
      route_table_ids.#: <computed>
      service_name:      "com.amazonaws.us-gov-west-1.s3"
      vpc_id:            "vpc-foovpcid"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

### Actual Behavior

`terraform plan` threw an error: 
```
> terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.aws_vpc_endpoint_service.s3: Refreshing state...

Error: Error refreshing state: 1 error(s) occurred:

* data.aws_vpc_endpoint_service.s3: 1 error(s) occurred:

* data.aws_vpc_endpoint_service.s3: data.aws_vpc_endpoint_service.s3: no matching VPC Endpoint Service found
```

### Steps to Reproduce
1. `terraform plan`

### PR DESCRIPTION
**Pull request by jmcarp**
Revert to logic from 1.8 for GovCloud until service details are
supported.

Resolves #3506.

Note: I haven't had time to test this thoroughly, so it would be great to get help testing this out. cc @lorengordon @bflad

### CODE REVIEW

**Comment by lorengordon** [comment]
I can knock out a test if someone wouldn't mind providing a windows binary... though I can probably compile one myself tomorrow.

File: aws/data_source_aws_vpc_endpoint_service.go [hunk 1]
```diff
@@ -85,6 +86,21 @@ func dataSourceAwsVpcEndpointServiceRead(d *schema.ResourceData, meta interface{
 	if err != nil {
 		return fmt.Errorf("Error fetching VPC Endpoint Services: %s", err)
 	}
+
+	// Note: AWS Commercial now returns a response with `ServiceNames` and
+	// `ServiceDetails`, but GovCloud responses only include `ServiceNames`
+	if isGovCloud && resp.ServiceDetails == nil {
```
Comment (bflad) [inline aws/data_source_aws_vpc_endpoint_service.go]: As written, this line can cause a panic if `resp == nil`

**Comment by bflad** [comment]
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by lorengordon** [comment]
@bflad Thanks, I just tested it and it looks like just going from v1.8.0 => v1.11.0 and changing nothing else at all it wants to destroy and recreate the s3 endpoint.

**Comment by lorengordon** [comment]
Ahh, looks like this is the line that's changing and causing the new resource:

```
vpc_endpoint_type:       "" => "Gateway" (forces new resource)
```

**Comment by bflad** [comment]
Sorry for any confusion, @lorengordon, this pull request was specifically for the `aws_vpc_endpoint` data source returning "no matches" due to incorrect logic. The mentioned `aws_vpc_endpoint` resource (not data source) returning `"" => "Gateway" (forces new resource)` in US Gov partition will be addressed in pull request #3317, which I will probably be merging today if looks good and releasing in v1.12.0 of the AWS provider.

**Comment by lorengordon** [comment]
@bflad No worries, thanks for following up. I just subscribed to that PR also!

**Pull request merged**